lib/Invoice.js

```diff
@@ -66,6 +68,8 @@ export class Invoice {
     assert(Array.isArray(this._options.items),
       'Valid Items array missing from invoice options')
 
+    assert(!(this._options.adjustmentInvoice === true && typeof this._options.adjustmentInvoiceNumber === 'undefined'), 'If adjustmentInvoice is true, adjustmentInvoiceNumber must be specified.');
```

Comment:
Unfortunately the [documentation](https://docs.szamlazz.hu/#xsd-scheme-compliance) does not really describe enough details about these fields. 

For me, it is more logical if we keep this package as simple as possible. What do you think if we introduce only the `adjustmentInvoiceNumber` as a constructor parameter. And if it is a valid string we set the `this._options.adjustmentInvoice` true. In this case the developers do not have to set both fields.

I think the following logic is more proper
```javascript
if (this._options.adjustmentInvoiceNumber !== null && this._options.adjustmentInvoiceNumber !== undefined) {
  assert(typeof this._options.adjustmentInvoiceNumber !== 'string', '"adjustmentInvoiceNumber" should be a string')
  assert(this._options.adjustmentInvoiceNumber.length > 0, '"adjustmentInvoiceNumber" should be minimum 1 character')

  this._options.adjustmentInvoice = true
}
```

Please also write tests and extend the `readme.md` too.

Reply:
Unfortunately, this part of the documentation is not clearly defined. Thank you for the example correction; it was a great idea. I made only this change:

assert(typeof` this._options.adjustmentInvoiceNumber === 'string', '"adjustmentInvoiceNumber" should be a `string')

I've completed the tests, tried them manually with invoices as well, and the cases were successful. Additionally, I've updated the README.md file. Is this acceptable, or should I change anything?

Reply:
Please add unit tests to the `tests/invoice.spec.js`. The following tests are needed
- `adjustmentInvoiceNumber` is `null`
- `adjustmentInvoiceNumber` is `undefined`
- `adjustmentInvoiceNumber` is empty `string`
- `adjustmentInvoiceNumber` is not `string`. You can test it with `Date`, `Number` or `Boolean` or with all the 3 :)
- `adjustmentInvoiceNumber` is a non empty `string`

I think, good idea to create a new test suite `describe` block in the `_generateXML` test suite.

Reply:
As requested, I added the unit tests to the tests/invoice.spec.js file. The tests cover the following scenarios:

- adjustmentInvoiceNumber null
- adjustmentInvoiceNumber is undefined
- adjustmentInvoiceNumber empty string
- adjustmentInvoiceNumber is not a string. The tests also apply to Date, Number and Boolean types.
- adjustmentInvoiceNumber is not an empty string

As suggested, I have created a new test suite within the _generateXML test suite that contains these tests.

Thanks very much for your guidance and patience; it was very helpful and I managed to implement the tests accordingly. 
What do you think, is this suitable?